cmd/kubeadm/app/cmd/init.go

```diff
@@ -559,3 +555,24 @@ func getDirectoriesToUse(dryRun bool, dryRunDir string, defaultPkiDir string) (s
 
 	return defaultPkiDir, kubeadmconstants.KubernetesDir, kubeadmconstants.GetStaticPodDirectory(), kubeadmconstants.KubeletRunDirectory, nil
 }
+
+// showJoinCommand prints the join command after all the phases in init have finished
+func showJoinCommand(i *initData, out io.Writer) error {
+	_, kubeConfigDir, _, _, err := getDirectoriesToUse(i.dryRun, i.dryRunDir, i.cfg.CertificatesDir)
+	if err != nil {
+		return errors.Wrap(err, "failed to get directories to use")
+	}
+
+	adminKubeConfigPath := filepath.Join(kubeConfigDir, kubeadmconstants.AdminKubeConfigFileName)
+
+	tokens := i.Tokens()
+
+	// Prints the join command, multiple times in case the user has multiple tokens
+	for _, token := range tokens {
+		if err := printJoinCommand(out, adminKubeConfigPath, token, i.skipTokenPrint); err != nil {
```

Comment:
initData already holds a method to get the kubeconfig path:
`adminKubeConfigPath` -> `i.KubeConfigPath()`
it can replace the code above :)
```
	_, kubeConfigDir, _, _, err := getDirectoriesToUse(i.dryRun, i.dryRunDir, i.cfg.CertificatesDir)
	if err != nil {
		return errors.Wrap(err, "failed to get directories to use")
	}
 	adminKubeConfigPath := filepath.Join(kubeConfigDir, kubeadmconstants.AdminKubeConfigFileName)
```

Reply:
Update